src/components/interactive-builder/interactive-builder.component.tsx

```diff
@@ -204,15 +209,134 @@ const InteractiveBuilder: React.FC<InteractiveBuilderProps> = ({
     [onSchemaChange, schema, t]
   );
 
+  const handleFormValidation = () => {
+    setResponses([]);
+
+    if (schema) {
+      const parsedForm =
+        typeof schema == "string" ? JSON.parse(schema) : schema;
+
+      for (let i = 0; i < parsedForm.pages.length; i++) {
+        for (let j = 0; j < parsedForm.pages[i].sections.length; j++) {
+          for (
+            let k = 0;
+            k < parsedForm.pages[i].sections[j].questions.length;
+            k++
+          ) {
+            const questionObject = parsedForm.pages[i].sections[j].questions[k];
+
+            if (questionObject.type === "obsGroup") {
+              questionObject.questions.forEach((question) => {
+                handleQuestionValidation(question);
+              });
+            }
+            handleQuestionValidation(questionObject);
+          }
+        }
+      }
+    } else {
+      console.log("Empty form!");
```

Comment:
According to our code formatting conventions, it's not a good practice for production-ready code to have console log statements. That being said, our code linter has been configured to reject those by default. Perhaps you intended to use a `console.warn()` or `console.error()` instead?

Reply:
This is well noted, perhaps I'll just display it somewhere on the UI as suggested by @ebambo